Clean MockedDependenciesRule

TurpIF opened the pull request.

Amokrane commented:
@TurpIF  changes look good to me. Let me know when I can review (resolve conflicts, quality gate).

TurpIF commented:
> @TurpIF changes look good to me. Let me know when I can review (resolve conflicts, quality gate).

@Amokrane: The quality gate fails because I moved the `CdbMock` class and Sonar doesn't get the good coverage rate because it is tested by `PubSdkApiIntegrationTest` class which is in the `publicher-sdk` module. This is the effect I mentioned in EE-1370 this morning.

I rebased, let's wait the end of the CI to verify if quality check is still the same.

TurpIF commented:
Yep I confirm that quality gate is red because of the coverage wrongly detected on `CdbMock`. On 75 uncovered lines, 67 are in `CdbMock`.

TurpIF commented on test-utils/src/main/java/com/criteo/publisher/mock/MockedDependenciesRule.java:
```diff
@@ -82,43 +85,80 @@
+  private TestResource getInMemoryResource() {
+    if (inMemoryResource == null) {
+      // Order of creation is important: resources are setUp in this order before every test and tearDown in the reverse
+      // order after. So setUp is:
```
? I do not understand your comment.
If it is about the line break, it is there because the line length is the max: 120 chars.
